xml/output/mediawiki.d

```diff
@@ -0,0 +1,352 @@
+						break;
+					case "local-link":
+						descendOther(format!"[[#%s|"(encodeHtmlEntities(n.attributes.aaGet("target-id"))), "]]");
+						break;
+					case "tabs":
+						if (state.listLevel == 0)
+						{
+							state.inTable = true;
+							tableStarted = false;
+							descendHtml(null);
+							if (outputting && tableStarted)
+							{
+								if (newlinesWritten == 0)
+									bufs[currentPageName].put("\n");
+								bufs[currentPageName].put("|}\n");
+								newlinesWritten = 1;
+							}
+						}
+						else
+							descendHtml(null);
+						break;
+					case "li":
+						listStillActive = true;
+						descendOther(format!"%s "("*".replicate(state.listLevel)), "");
+						if (outputting && newlinesWritten == 0)
+						{
+							bufs[currentPageName].put("\n");
+							++newlinesWritten;
+						}
+						break;
+					case "p":
+						if (state.listLevel != 0 && !listStillActive && newlinesWritten != 0)
+						{
+							bufs[currentPageName].formattedWrite!"%s:"("*".replicate(state.listLevel));
+						}
+						descendHtml(null);
+						if (state.listLevel == 0 && !state.inHeading && !state.inTable)
+						{
+							if (outputting && newlinesWritten < 2)
+							{
+								bufs[currentPageName].put("\n".replicate(2 - newlinesWritten));
+								newlinesWritten = 2;
+							}
+						}
+						else if (!state.inHeading && !state.inTable)
+						{
+							listStillActive = false;
+							if (outputting)
+								bufs[currentPageName].put("<br />");
+						}
+						else if (state.inTable && tableStarted)
+						{
+							if (outputting)
+							{
+								if (newlinesWritten == 0)
+									bufs[currentPageName].put("\n");
+								bufs[currentPageName].put("|-\n");
+								newlinesWritten = 1;
+							}
+						}
+						break;
+					case "col":
+						if (!state.inTable)
+							descendOther("", " ");
+						else
+						{
+							if (outputting && newlinesWritten == 0)
+							{
+								bufs[currentPageName].put("\n");
+							}
+							if (outputting && !tableStarted)
+							{
+								bufs[currentPageName].put("{|\n");
+								tableStarted = true;
+							}
+							state.inCell = true;
+							descendOther("|", "");
+							if (outputting && newlinesWritten == 0)
+							{
+								bufs[currentPageName].put("\n");
+								newlinesWritten = 1;
+							}
+						}
+							
+						break;
+					default:
+						throw new Exception("Unknown XML tag " ~ n.tag);
+				}
+
+				break;
+			}
+			case XmlNodeType.Text:
+				if (outputting)
+				{
+					if (state.inHeading && n.tag != "" && !headingWritten)
+					{
+						if (newlinesWritten < 2)
+						{
+							bufs[currentPageName].put("\n".replicate(2 - newlinesWritten));
+						}
+						bufs[currentPageName].put("=".replicate(headingLevel));
+						headingWritten = true;
+
+						if (state.inSubpageHeading && nextPageName == "")
+						{
+							nextPageName = replaceFirst(encodeHtmlEntities!false(n.tag), regex(r" \(.*$"), "");
```

Comment:
Construct the regular expression object only once:
```suggestion
							nextPageName = replaceFirst(encodeHtmlEntities!false(n.tag), re!` \(.*$`, "");
```
(+ `import ae.utils.regex;`)

Reply:
But in this case I would avoid using regular expressions completely (which are rather expensive).
```suggestion
							nextPageName = n.tag.findSplit(` (`)[0].encodeHtmlEntities!false();
```